### PR TITLE
drop hdrhistogram

### PR DESCRIPTION
**Pull request by richardstartin**
related to #2178 but means we stop using HdrHistogram. I dropped the granularity from nanoseconds to microseconds because it's wasteful for our use case.

### CODE REVIEW

**Review by tylerbenson — APPROVED** [review]

File: dd-trace-core/src/main/java/datadog/trace/core/monitor/Timer.java [hunk 1]
```diff
@@ -67,7 +66,7 @@ public void stop() {
 
   private void record(long now) {
     // if it's longer than 30s, we have bigger problems
-    histogram.recordValue(Math.min(now - start, THIRTY_SECONDS_AS_NANOS));
+    histogram.accept(Math.min(NANOSECONDS.toMicros(now - start), THIRTY_SECONDS_AS_MICROS));
```
Comment (tylerbenson) [inline dd-trace-core/src/main/java/datadog/trace/core/monitor/Timer.java:69]: I assume the thread safety concerns are already encapsulated?

Comment (richardstartin) [reply dd-trace-core/src/main/java/datadog/trace/core/monitor/Timer.java:69]: yes, indeed. `PackedHistogram` wasn't threadsafe either.

**Comment by richardstartin** [comment]
I thought this would just work but it seems I was wrong. Converting to draft for now.

**Review by mcculls — APPROVED** [review]

**Review by bantonsson — APPROVED** [review]

**Comment by richardstartin** [comment]
This will require some rework to work on JDK7

**Review by devinsba — APPROVED** [review]
Sounds reasonable to drop java 7 to me

**Review by bantonsson — APPROVED** [review]

**Review by tylerbenson — APPROVED** [review]

**Pull request merged**